raft/raft_test.go

```diff
@@ -1225,6 +1225,156 @@ func TestLeaderStepdownWhenQuorumLost(t *testing.T) {
 	}
 }
 
+func TestLeaderSupersedingWithCheckQuorum(t *testing.T) {
+	a := newTestRaft(1, []uint64{1, 2, 3}, 10, 1, NewMemoryStorage())
+	b := newTestRaft(2, []uint64{1, 2, 3}, 10, 1, NewMemoryStorage())
+	c := newTestRaft(3, []uint64{1, 2, 3}, 10, 1, NewMemoryStorage())
+
+	a.checkQuorum = true
+	b.checkQuorum = true
+	c.checkQuorum = true
+
+	nt := newNetwork(a, b, c)
+
+	// Prevent campaigning from b
```

Comment:
this case will break without line 1240

i think we need to let b's elapsedTimeout reach to electionTimeout so that it can vote for a,
and also if randomizedElectionTimeout is exactly electionTimeout(that's gonna happen if the random part of timeout is 0) b is gonna start a vote, it affects the behavior of this case